File: c/src/date.cpp
#include <iostream>
#include <ctime>
#include "date.h"
using namespace std;

/*
	Insert Method

	Arg: the value want to set
	Ret: If successfully set
*/
bool Date::setYear(int _){
	if(_>YEAR_MAX)
		return false;
	year = _;
	return true;
}

bool Date::setMonth(int _){
	if(_>MONTH_MAX)
		return false;
	month = _;
	return true;
}

bool Date::setDay(int _){
	if(_>DAY_MAX)
		return false;
	day = _;
	return true;
}

bool Date::setHour(int _){
	if(_>HOUR_MAX)
		return false;
	hour = _;
	return true;
}

bool Date::setMinute(int _){
	if(_>MINUTE_MAX)
		return false;
	minute = _;
	return true;
}

/*
	Get Method

	Ret: The corresponding value
*/
int Date::getYear() const{
	return year;
}

int Date::getMonth() const{
	return month;
}

int Date::getDay() const{
	return day;
}

int Date::getHour() const{
	return hour;
}

int Date::getMinute() const{
	return minute;
}

// ----- Other method -----
// Get the current time
void Date::Now(){
	time_t d = time(0);
	struct tm *now = localtime(&d);
	//this->show(now);
	this->setYear(now->tm_year+1900);
	this->setMonth(now->tm_mon+1);
	this->setDay(now->tm_mday);
	this->setHour(now->tm_hour);
	this->setMinute(now->tm_min);
}

/*
	Show the time information

	Arg: The tm structure defined in time.h
*/
void Date::show(struct tm* now){
	cout << "year: " << (now->tm_year+1900) << endl
		 << "month: " << (now->tm_mon+1) << endl
		 << "day: " << now->tm_mday << endl 
		 << "hour: " << now->tm_hour << endl
		 << "minute: " << now->tm_min << endl;
}

// ----- Overloadding operator -----
Date& Date::operator=(const Date &rhs) {
	setYear(rhs.getYear());
	setMonth(rhs.getMonth());
	setDay(rhs.getDay());
	setHour(rhs.getHour());
	setMinute(rhs.getMinute());
	return *this;  					// Return a reference to myself.
}

File: android/Ivy/app/src/main/java/edu/sunner/ivy/Date.java
package edu.sunner.ivy;

import java.util.Calendar;

/**
 * Date structure
 *
 * @author sunner
 * @since 9/4/16.
 */
public class Date {
    private int year;
    private int month;
    private int day;
    private int hour;
    private int minute;

    // ----- Insert method -----

    /**
     * Insert year value.
     *
     * @param value the value want to insert
     * @return if insert successfully
     */
    public boolean setYear(int value) {
        if (value > Constant.YEAR_MAX) {
            return false;
        }
        year = value;
        return true;
    }

    /**
     * Insert month value.
     *
     * @param value the value want to insert
     * @return if insert successfully
     */
    public boolean setMonth(int value) {
        if (value > Constant.MONTH_MAX) {
            return false;
        }
        month = value;
        return true;
    }

    /**
     * Insert day value.
     *
     * @param value the value want to insert
     * @return if insert successfully
     */
    public boolean setDay(int value) {
        if (value > Constant.DAY_MAX) {
            return false;
        }
        day = value;
        return true;
    }

    /**
     * Insert hour value.
     *
     * @param value the value want to insert
     * @return if insert successfully
     */
    public boolean setHour(int value) {
        if (value > Constant.HOUR_MAX) {
            return false;
        }
        hour = value;
        return true;
    }

    /**
     * Insert minute value.
     *
     * @param value the value want to insert
     * @return if insert successfully
     */
    public boolean setMinute(int value) {
        if (value > Constant.MINUTE_MAX) {
            return false;
        }
        minute = value;
        return true;
    }

    // ----- Get method -----
    public int getYear() {
        return year;

    }

    public int getMonth() {
        return month;

    }

    public int getDay() {
        return day;

    }

    public int getHour() {
        return hour;

    }

    public int getMinute() {
        return minute;

    }

    // ----- Other method -----

    /**
     * Set the value according to the current time.
     */
    public void now() {
        Calendar calendar = Calendar.getInstance();
        year = calendar.get(Calendar.YEAR);
        month = calendar.get(Calendar.MONTH) + 1;
        day = calendar.get(Calendar.DAY_OF_MONTH);
        hour = calendar.get(Calendar.HOUR);
        minute = calendar.get(Calendar.MINUTE);
    }

    /**
     * Word object to the string type.
     *
     * @return the string
     */
    public String toString() {
        String res = "";
        res = res + year + "\n";
        res = res + month + "\n";
        res = res + day + "\n";
        res = res + hour + "\n";
        res = res + minute;
        return res;
    }
}
